ui/app/contexts/metametrics.js

```diff
@@ -18,14 +19,71 @@ import { getEnvironmentType } from '../../../app/scripts/lib/util'
 import {
   sendMetaMetricsEvent,
   sendCountIsTrackable,
+  segmentWriteKey,
 } from '../helpers/utils/metametrics.util'
 
+const analytics = new Analytics(segmentWriteKey)
+
 export const MetaMetricsContext = createContext(() => {
   captureException(
     Error(`MetaMetrics context function was called from a react node that is not a descendant of a MetaMetrics context provider`),
   )
 })
 
+export const MixPanelContext = createContext(() => {
+  captureException(
+    Error(`MixPanel context function was called from a react node that is not a descendant of a MetaMetrics context provider`),
+  )
+})
+
+export function MixPanelProvider ({ children }) {
+  const metaMetricsId = useSelector((state) => state.metamask.metaMetricsId)
+
+  const mixPanelEvent = (eventName, properties) => {
+    if (metaMetricsId) {
+      analytics.track({
+        event: eventName,
+        properties,
+        userId: metaMetricsId,
```

Comment:
I guess this provider doesn't allow submitting "anonymous" events? We would want to set `anonymousId` in place of `userId` for anonymous events, e.g. our transaction events.